### PR TITLE
Added Filter functionality

### PR DESCRIPTION
**Pull request by ayush65**
Added Sort Functionality
-> HIgh to Low
-> Low To high 

Added rating filter

Added filter by category for grocery items

### CODE REVIEW

File: src/SideBar/SideBar.js [hunk 1]
```diff
@@ -1,63 +1,152 @@
...
+          onClick={() =>
+            Dispatch({ type: "SORT_FILTER", payload: { sort: "DSC" } })
+          }
+          className="radio"
+        >
+          <input
+            type="radio"
+            value="highToLow"
+            name="price"
+            checked={sort === "DSC"}
```
Comment (narharikale) [inline src/SideBar/SideBar.js:33]: DSC , name could be better

File: src/SideBar/SideBar.js [hunk 2]
```diff
@@ -1,63 +1,152 @@
...
+          />
+          <span className="radio_text">Price- High to low</span>
+        </label>
+          
+            </li>
+              <li>
+        <label
+          onClick={() =>
+            Dispatch({ type: "SORT_FILTER", payload: { sort: "ASC" } })
```
Comment (narharikale) [inline src/SideBar/SideBar.js:42]: we can use constants instead of  SORT_FILTER , to avoid any mistake

File: src/SideBar/SideBar.js [hunk 3]
```diff
@@ -66,37 +155,156 @@ function SideBar() {
...
+          <span className="chechbox_text">breakfast Essential</span>
+        </label>
         </li>
-              
+  
+
         <li>
-        <input type="checkbox" className="sidebar-price-ranging price-range-btw-2000-2500" name="price-ranging" value="price-range"/>
-        <label >Cold drinks and juices</label>
+        <label for="Canvas" className="checkbox">
```
Comment (narharikale) [inline src/SideBar/SideBar.js:251]: in jsx we use htmlFor instead of for

File: src/context/filter-context.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    price: 50,
+    category: [],
+    rating: 5,
+    sort: null,
+    default: [],
+  });
+  useEffect(() => {
+    (async () => {
+      let response = await axios.get("/api/products");
+      console.log(response.data.products);
```
Comment (narharikale) [inline src/context/filter-context.js:128]: console here

File: src/context/filter-context.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+const filterReducer = (filterState, action) => {
```
Comment (narharikale) [inline src/context/filter-context.js:3]: can we extract this function out, so it  will much readable

**Review by narharikale — APPROVED** [review]
- you can improve folder structure
- I can see there is scope to improve code readability

File: src/context/filter-context.js (hunk 5 above)
Comment (ayush65) [reply src/context/filter-context.js]: yea i have  done the changes you have suggested  . Thanks for reviewing

**Review by kotesh-arya — APPROVED** [review]
Just follow the improvements suggested by the other folk
rest of the code looks good and well defined 👍️

File: src/context/filter-context.js (hunk 4 above)
Comment (Himanshu-Dhawale) [inline src/context/filter-context.js]: You can remove the console.log part here

**Review by Himanshu-Dhawale — APPROVED** [review]
UI is good and the code is also clean. Just suggested a little change

File: src/context/filter-context.js (hunk 4 above)
Comment (ayush65) [reply src/context/filter-context.js]: yea i have dne the changes . thanks for reviewing

**Pull request merged**